src/memory.rs

```diff
@@ -120,7 +120,7 @@ impl<'a, 'tcx> Memory<'a, 'tcx> {
             bytes: Vec::new(),
             relocations: BTreeMap::new(),
             undef_mask: UndefMask::new(0),
-            align: 1,
+            align: 8, // should be infinity?
```

Comment:
Alignment is irrelevant for ZSTs. miri *must* treat reads/writes of ZSTs as noops and do *absolutely nothing*.